src/library/scala/collection/mutable/LongMap.scala

```diff
@@ -682,4 +677,11 @@ object LongMap {
 
   implicit def iterableFactory[V]: Factory[(Long, V), LongMap[V]] = toFactory(this)
   implicit def buildFromLongMap[V]: BuildFrom[LongMap[_], (Long, V), LongMap[V]] = toBuildFrom(this)
+
+  private def repackMask(mask: Int, _size: Int, _vacant: Int): Int = {
```

Comment:
I don't think it is good practice to factor out code to a companion object simply because there was a bug in the code and we want to test the behavior in isolation.  It's easier to understand in context.  I'd leave it where it was.

Reply:
I did not trouble myself to understand the fix, so I'm agnostic on that point. But I think it's normal to factor out "I need to calculate a thing", especially if the experts disagree on what to calculate.

But in this case, I would prefer a unit test that constructs the data structure to elicit the bug. I don't know if the putative "bad state" is still a possible state.

Reply:
I disagree, I think it is a good idea to factor out the mask computation in another function. Also, when triggered, the bug results in an infinite loop, which isn't practical for a unit test.

Reply:
If there were a regression in the factored-out function, it would cause an infinite loop there, too.

The difference is that you don't have to spend 20-30s of computation time to build a big enough LongMap.

But, anyway, I don't care about this particularly strongly.

Reply:
Not if the regression is in the same vein: here the bug just makes `m` too small, the infinite loop happens when the key-value pairs are added in the very small new array.
But I agree that you can make this loop infinite in some way.